kiota_abstractions/serialization/untyped_dict.py

```diff
@@ -0,0 +1,24 @@
+from .untyped_node import UntypedNode
+
+class UntypedDictionary(UntypedNode):
```

Comment:
```suggestion
class UntypedObject(UntypedNode):
```
Is there any reason to not call it `UntypedObject`?

Reply:
Initially, I though of adding it as `UntypedObject` but in python we call it dictionary which stores key-value pairs similar to `HashMap` in `Java`. https://docs.python.org/3/tutorial/datastructures.html#dictionaries

I can change it back to `Object` if you like.

And, yes I will look into serialization implementation as well, but first I need to get my head around it.

Thanks

Reply:
I don't have strong opinion on this one. It's better to align with what the language thinks is best. 
Maybe @shemogumbe can weigh in on the naming here..

Reply:
My view is - `arrays `and `objects ` instead of `list` and `dict` might create confusion as the codebase of the project using python client will have lists and dictionaries. 

open for suggestions.

Reply:
@rohitkrsoni Lets keep this as is for now since the it makes sense for the language to have the naming as so and then start working on the serialization bits. In the event of any feedback before we merge, we can do a renaming.